CMakeLists.txt

```diff
@@ -15,44 +15,91 @@ option(WITH_OTLP "Whether to include the OpenTelemetry Protocol in the SDK" OFF)
 option(WITH_PROMETHEUS "Whether to include the Prometheus Client in the SDK"
        OFF)
 
-option(WITH_TESTS "Whether to enable tests" ON)
+option(BUILD_TESTING "Whether to enable tests" ON)
 option(WITH_EXAMPLES "Whether to build examples" ON)
 
 set(WITH_PROTOBUF OFF)
-if(WITH_OTLP)
-  set(WITH_PROTOBUF ON)
-endif()
-
-if(WITH_TESTS)
-  include(CTest)
-endif()
 
 find_package(Threads)
 
+function(install_windows_deps)
+  # Bootstrap vcpkg from CMake and auto-install deps in case if we are missing
+  # deps on Windows
+  message("Installing build tools and dependencies...")
+  execute_process(
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/tools/setup-buildtools.cmd)
+  set(CMAKE_TOOLCHAIN_FILE
+      ${CMAKE_CURRENT_SOURCE_DIR}/tools/vcpkg/scripts/buildsystems/vcpkg.cmake)
+endfunction()
+
 if(MSVC)
   # Options for Visual C++ compiler: /Zc:__cplusplus - report an updated value
   # for recent C++ language standards. Without this option MSVC returns the
   # value of __cplusplus="199711L"
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus")
 endif()
 
+if(WITH_OTLP)
+  set(WITH_PROTOBUF ON)
+endif()
+
 if(WITH_PROTOBUF)
   set(protobuf_MODULE_COMPATIBLE ON)
-  find_package(Protobuf CONFIG NAMES protobuf)
-  # Older versions of protobuf don't use cmake config files.
```

Comment:
We can't use old versions anyways, since the `opentelemetry-proto` REQUIRES at least proto3...